### ISSUE TITLE
ArgumentNullException when starting endpoint

### ISSUE THREAD

**Issue by robbaman** [Particular/NServiceBus#5551]
I'm running into a bit of an odd situation where starting the endpoint immediately throws the following ArgumentNullException:

```
System.ArgumentNullException: Value cannot be null. (Parameter 'name')
   at System.Type.GetProperty(String name, BindingFlags bindingAttr)
   at NServiceBus.ConcreteProxyCreator.BuildCustomAttribute(Object customAttribute)
   at NServiceBus.ConcreteProxyCreator.CreateTypeFrom(Type type)
   at NServiceBus.MessageInterfaces.MessageMapper.Reflection.MessageMapper.GenerateImplementationFor(Type interfaceType)
   at NServiceBus.MessageInterfaces.MessageMapper.Reflection.MessageMapper.InnerInitialize(Type t)
   at NServiceBus.MessageInterfaces.MessageMapper.Reflection.MessageMapper.InitType(Type t)
   at NServiceBus.MessageInterfaces.MessageMapper.Reflection.MessageMapper.Initialize(IEnumerable`1 types)
   at NServiceBus.SerializationFeature.Setup(FeatureConfigurationContext context)
   at NServiceBus.Features.FeatureActivator.FeatureInfo.InitializeFrom(FeatureConfigurationContext featureConfigurationContext)
   at NServiceBus.Features.FeatureActivator.ActivateFeature(FeatureInfo featureInfo, List`1 featuresToActivate, FeatureConfigurationContext featureConfigurationContext)
   at NServiceBus.Features.FeatureActivator.SetupFeatures(FeatureConfigurationContext featureConfigurationContext)
   at NServiceBus.FeatureComponent.Initalize(ContainerComponent containerComponent, FeatureConfigurationContext featureConfigurationContext)
   at NServiceBus.EndpointCreator.Initialize()
   at NServiceBus.EndpointCreator.CreateWithInternallyManagedContainer(EndpointConfiguration endpointConfiguration)
   at NServiceBus.Endpoint.Create(EndpointConfiguration configuration)
   at NServiceBus.Endpoint.Start(EndpointConfiguration configuration)
```

I've narrowed it down to the following command specification which if present like below causes the exception:

```
public interface ISendEmailCommand : ICommand {
	List<SendEmailCommandAddress> Recipients { get; set; }
	string? Subject { get; set; }
	string? BodyHtml { get; set; }
}

public class SendEmailCommandAddress {
	public SendEmailCommandAddress(string emailAddress) {
		EmailAddress = emailAddress;
	}
	public string EmailAddress { get; set; }
	public string? Name { get; set; }
}
```

This uses the new C#8 nullable type feature for both the interface as well as the class. **It appears that NServiceBus crashes on the nullable type on the interface.** Changing the code to this does work:

```
public interface ISendEmailCommand : ICommand {
	List<SendEmailCommandAddress> Recipients { get; set; }
	string Subject { get; set; }
	string BodyHtml { get; set; }
}

public class SendEmailCommandAddress {
	public SendEmailCommandAddress(string emailAddress) {
		EmailAddress = emailAddress;
	}
	public string EmailAddress { get; set; }
	public string? Name { get; set; }
}
```

Note that the class still has the *string?* type which works fine, but as soon as the interface gets a nullable reference type property the ArgumentNullException occurs during startup.

**Comment by andreasohlund** [comment]
@robbaman this has been fixed in https://github.com/Particular/NServiceBus/pull/5488 but not yet released, what's the urgency for getting you a fix?

**Comment by robbaman** [comment]
Not at all, we just started using the nullable reference types feature recently and thus were not aware this was an issue. Checked but didn't find an open issue so figured I'd report it. For now I'll just not make the properties nullable.

Thanks

**Comment by andreasohlund** [comment]
@robbaman got it, we'll roll this into the next minor. Thanks for taking the time to report it 👍 

Closing this as fixed by #5488

**Comment by robbaman** [comment]
It turns out we are using this somewhat more intensively on a different project. Do you have an ETA on the next minor release? I'd hate to have to change all our string?'s to string only to have to change it back a week later.

**Comment by andreasohlund** [comment]
@robbaman we have a patch coming up in the next 1-2 weeks that would contain this. Is that timeframe acceptable to you? (I not I can give you access to a prerelease build in the meantime)

**Comment by robbaman** [comment]
@andreasohlund if it's not too difficult it'd be great to get access to a prerelease build. We won't go live within the next few weeks but seeing as I can't start the endpoint without either the new version or changing most of my interfaces it would help me with testing and development.

Thanks.

**Comment by andreasohlund** [comment]
https://www.myget.org/feed/particular/package/nuget/NServiceBus/7.3.0-beta0059 should contain the fix

Note that the fix will be pulled into a `7.2.1` release but the above should work for now. Let me know if this works?

**Comment by robbaman** [comment]
It does indeed. Thanks for the quick response.
